lib/index.js

```diff
@@ -40,7 +40,8 @@ exports.register = (server, options, next) => {
   });
 
   server.ext('onPreResponse', (request, reply) => {
-    const rate = request.plugins['hapi-rate-limit'].rate;
+
+    const rate = request.plugins['hapi-rate-limiter'] ? request.plugins['hapi-rate-limiter'].rate : null;
```

Comment:
In the case that a request does not pass auth, the `onPostAuth` handler will not be called, where the state for `request.plugins['hapi-rate-limiter']` is set. However, `onPreResponse` will still be called, and without this check, will try to access the `rate` property when it hasn't been set. This guards against that.